baseapp/middleware_test.go

```diff
@@ -0,0 +1,1359 @@
+package baseapp_test
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"math"
+	"math/rand"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/gogo/protobuf/jsonpb"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/libs/log"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	dbm "github.com/tendermint/tm-db"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/snapshots"
+	snapshottypes "github.com/cosmos/cosmos-sdk/snapshots/types"
+	"github.com/cosmos/cosmos-sdk/testutil/testdata"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/types/tx"
+
+	"github.com/cosmos/cosmos-sdk/x/auth/middleware"
+	"github.com/cosmos/cosmos-sdk/x/auth/migrations/legacytx"
+)
+
+var (
+	capKey1 = sdk.NewKVStoreKey("key1")
+	capKey2 = sdk.NewKVStoreKey("key2")
+
+	interfaceRegistry = testdata.NewTestInterfaceRegistry()
+)
+
+type paramStore struct {
```

Comment:
I noticed there are a lot of duplicate code between baseapp_test.go and middleware_test.go (paramStore, setupBaseApp...).

Ideally, a clean separation would be:
- all tests that need a TX should go in middleware_test.go
- all other test should stay in baseapp_test.go

It might be a bit tricky with dep cycles and _test not _test package to share code. LMK if you need any help with that, I can also try to think about this.